types/block.go

```diff
@@ -817,6 +961,51 @@ func (sh SignedHeader) StringIndented(indent string) string {
 		indent)
 }
 
+// ToProto converts SignedHeader to protobuf
+func (sh *SignedHeader) ToProto() *tmproto.SignedHeader {
+	if sh == nil {
+		return nil
+	}
+
+	psh := new(tmproto.SignedHeader)
+	if sh.Header != nil {
```

Comment:
this is not possible imho (if Tendermint works correctly)

Reply:
I do think having the case may be good. It may not be possible but to test this func I do need to check this